var/spack/repos/builtin/packages/python/package.py

```diff
@@ -1122,82 +853,317 @@ def setup_dependent_build_environment(self, env, dependent_spec):
+                else:
+                    self.pkg.command("-c", "import tkinter")
+
+            # Ensure that uuid module works
+            if "+uuid" in spec:
+                self.pkg.command("-c", "import uuid")
+
+            # Ensure that tix module works
+            if "+tix" in spec:
+                self.pkg.command("-c", "import tkinter.tix")
+
+            # Ensure that crypt module works
+            if "+crypt" in spec:
+                self.pkg.command("-c", "import crypt")
+
+
+class AutotoolsBuilder(autotools.AutotoolsBuilder, RunAfter, BuildEnvironment):
+    # An in-source build with --enable-optimizations fails for python@3.X
+    build_directory = "spack-build"
+
+    def configure_args(self):
+        spec = self.spec
+        config_args = []
+        cflags = []
+
+        # setup.py needs to be able to read the CPPFLAGS and LDFLAGS
+        # as it scans for the library and headers to build
+        link_deps = spec.dependencies(deptype="link")
+
+        if link_deps:
+            # Header files are often included assuming they reside in a
+            # subdirectory of prefix.include, e.g. #include <openssl/ssl.h>,
+            # which is why we don't use HeaderList here. The header files of
+            # libffi reside in prefix.lib but the configure script of Python
+            # finds them using pkg-config.
+            cppflags = " ".join("-I" + spec[dep.name].prefix.include for dep in link_deps)
+
+            # Currently, the only way to get SpecBuildInterface wrappers of the
+            # dependencies (which we need to get their 'libs') is to get them
+            # using spec.__getitem__.
+            ldflags = " ".join(spec[dep.name].libs.search_flags for dep in link_deps)
+
+            config_args.extend(["CPPFLAGS=" + cppflags, "LDFLAGS=" + ldflags])
+
+        if "+optimizations" in spec:
+            config_args.append("--enable-optimizations")
+            # Prefer thin LTO for faster compilation times.
+            if "@3.11.0: %clang@3.9:" in spec or "@3.11.0: %apple-clang@8:" in spec:
+                config_args.append("--with-lto=thin")
             else:
-                os.remove(dst)
+                config_args.append("--with-lto")
+            config_args.append("--with-computed-gotos")
 
-    def test(self):
-        # do not use self.command because we are also testing the run env
-        exe = self.spec["python"].command.name
+        if spec.satisfies("@3.7 %intel"):
+            config_args.append("--with-icc={0}".format(spack_cc))
 
-        # test hello world
-        msg = "hello world!"
-        reason = "test: running {0}".format(msg)
-        options = ["-c", 'print("{0}")'.format(msg)]
-        self.run_test(exe, options=options, expected=[msg], installed=True, purpose=reason)
+        if "+debug" in spec:
+            config_args.append("--with-pydebug")
+        else:
+            config_args.append("--without-pydebug")
```

Comment:
Here you can simplify a lot of boilerplate (beware of typos, I wrote this in the UI :slightly_smiling_face: ):
```suggestion
        config_args.extend(self.with_or_without("pydebug", variant="debug"))
```